mono/metadata/threadpool-io.c

```diff
@@ -317,12 +317,22 @@ static gsize WINAPI
 selector_thread (gpointer data)
 {
 	MonoError error;
+	error_init (&error);
 	MonoGHashTable *states;
 
-	MonoString *thread_name = mono_string_new_checked (mono_get_root_domain (), "Thread Pool I/O Selector", &error);
+#if 0 // Thread name cannot be set here in the usual way.
+	MonoStringHandle thread_name = mono_string_new_handle (mono_get_root_domain (), "Thread Pool I/O Selector", &error);
 	mono_error_assert_ok (&error);
-	mono_thread_set_name_internal (mono_thread_internal_current (), thread_name, FALSE, TRUE, &error);
+	mono_thread_set_name_handle_internal (mono_thread_internal_current (), thread_name, FALSE, TRUE, &error);
 	mono_error_assert_ok (&error);
+#else
+#define SELECTOR_THREAD_NAME 'T','h','r','e','a','d',' ','P','o','o','l',' ','I','/','O',' ','S','e','l','e','c','t','o','r',0
+	const static char selector_thread_name_utf8[] = {SELECTOR_THREAD_NAME};
+	const static mono_unichar2 selector_thread_name_utf16[] = {SELECTOR_THREAD_NAME};
+	mono_thread_set_name_utf_internal (mono_thread_internal_current (), sizeof (selector_thread_name_utf8) - 1,
+			selector_thread_name_utf8, selector_thread_name_utf16, FALSE, TRUE, &error);
+	mono_error_assert_ok (&error);
```

Comment:
`#undef SELECTOR_THREAD_NAME`.

But again, I'm unclear on what the problem is with the straightforward method.

Reply:
ok. This one I don't remember the problem a test (make check) clearly didn't like it. I'd just as soon make all them use the utf function and skip managed.